feat(core): respect configurable agent ignore files (#4653)
QwenLM/qwen-code: docs/users/configuration/settings.md

| `context.fileFiltering.respectGitIgnore`                 | boolean                    | Respect .gitignore files when searching.                                                                                                                                                                                                                                                                                                                              | `true`                          |
| `context.fileFiltering.respectQwenIgnore`                | boolean                    | Respect .qwenignore and configured custom ignore files when searching.                                                                                                                                                                                                                                                                                                | `true`                          |

> | `context.fileFiltering.customIgnoreFiles`                | array                      | Additional project-root-relative ignore files to respect when `respectQwenIgnore` is enabled. `.qwenignore` is always included; this setting defaults to compatibility files `.agentignore` and `.aiignore`.                                                                                                                                                          | `[".agentignore", ".aiignore"]` |

Reviewer: **[Suggestion]** "**Additional** project-root-relative ignore files" implies additive semantics (values added on top of defaults), but the implementation has **replacement** semantics — `Config` uses `params.fileFiltering?.customIgnoreFiles ?? [...DEFAULT]`, so any provided value fully replaces the defaults.

`settingsSchema.ts` correctly says "to use **instead of** the defaults". This documentation contradicts it.

```suggestion
| `context.fileFiltering.customIgnoreFiles`                | array                      | Project-root-relative ignore files to use instead of the default compatibility files (`.agentignore`, `.aiignore`) when `respectQwenIgnore` is enabled. `.qwenignore` is always included.                                                                                                                                                          | `[".agentignore", ".aiignore"]` |
```

_— qwen3.7-max via Qwen Code /review_
